File: trtf_build/trtf_build/runtime_config/schemas/audio_bark.py
"""Schema for the ``audio_bark`` namespace.

Per-session sampling / debug knobs for the Bark TTS pipeline. Replaces
``TRTF_BARK_DUMP``, ``TRTF_BARK_GREEDY``, ``TRTF_BARK_SEED``.

Using ``audio_bark`` (underscore) instead of ``audio.bark`` because the
current registry keys the namespace on a single string without dotted
sub-paths — matching how ``triattention``, ``decode_policy``, etc. work.
"""

from __future__ import annotations

from trtf_build.runtime_config import (
    ConfigField,
    Layer,
    Schema,
    register_schema,
)


_SESSION = frozenset({Layer.SESSION_REQUEST, Layer.PLATFORM_PROFILE})


SCHEMA = Schema(
    namespace="audio_bark",
    fields=(
        ConfigField(
            name="dump_path",
            type_tag="string",
            default="",  # empty -> no dump
            allowed_layers=_SESSION,
        ),
        ConfigField(
            name="greedy",
            type_tag="bool",
            default=False,
            allowed_layers=_SESSION,
        ),
        ConfigField(
            name="seed",
            type_tag="int64",
            default=-1,  # -1 -> use default RNG state
            allowed_layers=_SESSION,
        ),
    ),
)


register_schema(SCHEMA)

File: src/runtime/config/schemas/audio_bark.cpp
// Registration for the "audio_bark" namespace schema.
// Mirrors trtf_build/trtf_build/runtime_config/schemas/audio_bark.py.

#include "trtf/config/schemas/audio_bark.h"

#include <any>
#include <cstdint>
#include <set>
#include <string>

namespace trtf::config::schemas {

Schema make_audio_bark_schema() {
    const std::set<Layer> session = {Layer::SessionRequest, Layer::PlatformProfile};
    return Schema{
        "audio_bark",
        {
            ConfigField{"dump_path", "string", std::any{std::string{}}, session, nullptr},
            ConfigField{"greedy", "bool", std::any{false}, session, nullptr},
            ConfigField{"seed", "int64", std::any{std::int64_t{-1}}, session, nullptr},
        },
    };
}

REGISTER_CONFIG_SCHEMA_FACTORY_WITH_FORCE_LINK(kForceLink_audio_bark, make_audio_bark_schema);
} // namespace trtf::config::schemas
